src/internal/Observable.ts

```diff
@@ -500,13 +494,9 @@ function getPromiseCtor(promiseCtor: PromiseConstructorLike | undefined) {
 export function canReportError(subscriber: Subscriber<any>): boolean {
   while (subscriber) {
     const { closed, destination, isStopped } = subscriber as any;
-    if (closed || isStopped) {
-      return false;
-    } else if (destination && destination instanceof Subscriber) {
-      subscriber = destination;
-    } else {
-      subscriber = null!;
-    }
+    return closed || isStopped
```

Comment:
This return changes makes the `while` obsolete; I'm wondering if this change is correct as it changes the semantics?

I think it could be this:

```ts
for (;;) {
  const { closed, destination, isStopped } = subscriber as any;
  if (closed || isStopped) {
    return false;
  }
  if (destination instanceof Subscriber) {
    subscriber = destination;
  } else {
    break;
  }
}
```

Reply:
Well, the guy that did that was tired. 😛 

Haha... thanks for catching that @JoostK .. although it's disconcerting that wasn't found in tests. Updated.